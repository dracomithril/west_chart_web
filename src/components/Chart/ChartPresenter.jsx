/**
 * Created by Gryzli on 06.04.2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { Tab, Tabs } from 'react-bootstrap';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import { faFacebook, faSpotify } from '@fortawesome/fontawesome-free-brands';
import { faList, faTable } from '@fortawesome/fontawesome-free-solid';

import Summary from '../Summary';
import ChartTable from './ChartTable';
import ChartHeader from './ChartHeader';
import SpotifySearch from '../Playlist/SpotifySearch';
import WestLetter from '../WestLetter';
import { filterChart, sorting } from '../../utils/utils';
import UpdateInfo from './UpdateInfo';

const ChartPresenter = (props, { store }) => {
  const { list_sort, chart, filters, until, songs_per_day, since, last_update } = store.getState();
  const defaultValue = {
    view_chart: [],
    error_days: null,
    westLetters: [],
  };
  const { view_chart, error_days, westLetters } =
    chart.length > 0 ? filterChart(chart, filters, until, songs_per_day) : defaultValue;
  const selected = view_chart.filter(elem => elem.selected);
  sorting[list_sort](selected);
  const show = !!last_update && !!since && !!until;
  return (
    <div>
      <Tabs defaultActiveKey={0} id="chart_tabs">
        <Tab
          eventKey={0}
          title={
            <div>
              <FontAwesomeIcon icon={faFacebook} />Posts
            </div>
          }
        >
          <div className="chart-presenter__tab-content">
            <ChartHeader error_days={error_days} view_chart={view_chart} />
            <div>
              {show && (
                <UpdateInfo
                  last_update={last_update}
                  until={until}
                  since={since}
                  total={chart.length}
                  filtered={view_chart.length}
                />
              )}
            </div>
            <ChartTable data={view_chart} />
          </div>
        </Tab>
        <Tab
          eventKey={1}
          title={
            <div id="chart_playlist_tab">
              <FontAwesomeIcon icon={faSpotify} />
              Playlist
            </div>
          }
        >
          <SpotifySearch selected={selected} />
        </Tab>
        <Tab
          eventKey={2}
          title={
            <div>
              <FontAwesomeIcon icon={faList} />Summary
            </div>
          }
        >
          <Summary selected={selected} />
        </Tab>
        <Tab
          eventKey={3}
          title={
            <div>
              <FontAwesomeIcon icon={faTable} />West Letters
            </div>
          }
        >
          <WestLetter data={westLetters} />
        </Tab>
      </Tabs>
    </div>
  );
};
ChartPresenter.contextTypes = {
  store: PropTypes.object,
};
export default ChartPresenter;
