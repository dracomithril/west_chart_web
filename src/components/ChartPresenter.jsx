/**
 * Created by Gryzli on 06.04.2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { Tab, Tabs } from 'react-bootstrap';
import Summary from './Summary';
import ChartTable from './ChartTable';
import ChartHeader from './ChartHeader';
import SpotifySearch from './SpotifySearch';
import WestLetter from './WestLetter';
import { filterChart, sorting } from '../utils/utils';

const ChartPresenter = (props, { store }) => {
  const { list_sort, chart, filters, until, songs_per_day } = store.getState();
  const defaultValue = {
    view_chart: [],
    error_days: [],
    westLetters: [],
  };
  const { view_chart, error_days, westLetters } =
    chart.length > 0 ? filterChart(chart, filters, until, songs_per_day) : defaultValue;
  const selected = view_chart.filter(elem => elem.selected);
  sorting[list_sort](selected);
  return (
    <div>
      <Tabs defaultActiveKey={0} id="chart_tabs">
        <Tab eventKey={0} title={<i className="fab fa-facebook">Posts</i>}>
          <ChartHeader error_days={error_days} view_chart={view_chart} />
          <div>
            We did get <span>{chart.length}</span> and filtered <span>{view_chart.length}</span>
          </div>
          <ChartTable data={view_chart} />
        </Tab>
        <Tab
          eventKey={1}
          title={
            <i className="fab fa-spotify" id="chart_playlist_tab">
              Playlist
            </i>
          }
        >
          <SpotifySearch selected={selected} />
        </Tab>
        <Tab eventKey={2} title={<i className="fas fa-list">Summary</i>}>
          <Summary selected={selected} />
        </Tab>
        <Tab eventKey={3} title={<i className="fas fa-table">West Letters</i>}>
          <WestLetter data={westLetters} />
        </Tab>
      </Tabs>
    </div>
  );
};
ChartPresenter.contextTypes = {
  store: PropTypes.object,
};
export default ChartPresenter;
