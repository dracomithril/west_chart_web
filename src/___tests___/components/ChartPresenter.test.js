/**
 * Created by XKTR67 on 4/19/2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import ChartPresenter from './../../components/ChartPresenter';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';

const initial_state = require('../data/initial_state.json');

Enzyme.configure({ adapter: new Adapter() });
const mockStore = configureMockStore([]);
describe('<ChartPresenter/>', () => {
  it('renders without crashing ChartPresenter', () => {
    const store = mockStore(initial_state);
    const wrapper = shallow(
      <ChartPresenter view_chart={[]} />, {
        context: { store },
        childContextTypes: { store: PropTypes.object },
      },
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });

});
