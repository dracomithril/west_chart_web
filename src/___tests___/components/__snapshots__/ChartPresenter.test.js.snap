// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartPresenter/> renders without crashing ChartPresenter 1`] = `
<div>
  <Uncontrolled(Tabs)
    defaultActiveKey={0}
    id="chart_tabs"
  >
    <Tab
      eventKey={0}
      title={
        <i
          className="fa fa-facebook"
        >
          Posts
        </i>
      }
    >
      <ChartHeader
        error_days={Array []}
        view_chart={Array []}
      />
      <ChartTable
        data={Array []}
      />
    </Tab>
    <Tab
      eventKey={1}
      title={
        <i
          className="fa fa-spotify"
          id="chart_playlist_tab"
        >
          Playlist
        </i>
      }
    >
      <SpotifySearch
        selected={Array []}
      />
    </Tab>
    <Tab
      eventKey={2}
      title={
        <i
          className="fa fa-list"
        >
          Summary
        </i>
      }
    >
      <Summary
        selected={Array []}
      />
    </Tab>
    <Tab
      eventKey={3}
      title={
        <i
          className="fa fa-table"
        >
          West Letters
        </i>
      }
    >
      <WestLetter
        data={Array []}
      />
    </Tab>
  </Uncontrolled(Tabs)>
</div>
`;
